x-pack/legacy/plugins/index_management/public/app/components/mappings_editor/components/document_fields/fields/field_types/completion_type.tsx

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React from 'react';
+
+import { i18n } from '@kbn/i18n';
+
+import { EuiSpacer } from '@elastic/eui';
+import { NormalizedField, Field as FieldType } from '../../../../types';
+import { getFieldConfig } from '../../../../lib';
+import { UseField, Field, CheckBoxField, FormDataProvider } from '../../../../shared_imports';
+import { AnalyzerParameter } from '../../field_parameters';
+import { EditFieldSection, EditFieldFormRow, AdvancedSettingsWrapper } from '../edit_field';
+
+const getDefaultValueToggle = (param: string, field: FieldType) => {
+  switch (param) {
+    case 'max_input_length': {
+      return field[param] !== undefined && field[param] !== getFieldConfig(param).defaultValue;
+    }
+    case 'analyzers': {
+      return field.search_analyzer !== undefined && field.search_analyzer !== field.analyzer;
+    }
+    default:
+      return false;
+  }
+};
+
+interface Props {
+  field: NormalizedField;
+}
+
+export const CompletionType = ({ field }: Props) => {
```

Comment:
I think I would at least have some settings outside the Advanced ones. WDYT?
<img width="701" alt="Screen Shot 2019-11-21 at 19 46 50" src="https://user-images.githubusercontent.com/2854616/69345845-d61c5f00-0c97-11ea-9a4e-0a5d6388dec0.png">

Reply:
I'd think I'd like to defer to the ES team and get their feedback. If it makes sense to pull some of them out of Advanced, I'm ok with that, but I don't know enough about this so I'd rather not arbitrarily choose some :)

Reply:
Ok My point of view is that, if there are _advanced_ settings, there must be other settings (_basic_). Otherwise, they are not "advanced" 😊 In that case, it should be called "Show settings". And then I wonder, why force the user to make another click to change a setting?

Reply:
I'm going to leave as is for now, and will revisit once I get feedback from the ES team.